protocol/communities/community.go

```diff
@@ -1690,17 +1697,18 @@ func dehydrateChannelsMembers(description *protobuf.CommunityDescription) {
 	for channelID, channel := range description.Chats {
 		if !channelHasPermissions(ChatID(description.ID, channelID), description.TokenPermissions) {
 			channel.Members = map[string]*protobuf.CommunityMember{} // clean members
-		}
-	}
-}
-
-func hydrateChannelsMembers(description *protobuf.CommunityDescription) {
-	for channelID, channel := range description.Chats {
-		if !channelHasPermissions(ChatID(description.ID, channelID), description.TokenPermissions) {
-			channel.Members = make(map[string]*protobuf.CommunityMember)
-			for pubKey, member := range description.Members {
-				channel.Members[pubKey] = member
+		} else if !channelEncrypted(ChatID(description.ID, channelID), description.TokenPermissions) {
+			// Has some permissions, but not encrypted. Everyone can view, only members can post.
+			membersToKeep := make(map[string]*protobuf.CommunityMember)
+			for id, member := range channel.Members {
+				isViewer := member.GetChannelRole() == protobuf.CommunityMember_CHANNEL_ROLE_VIEWER
+				if !isViewer {
+					// If the member is a viewer, we can remove them from the channel members
+					// as they are not allowed to post anyway.
+					membersToKeep[id] = member
+				}
 			}
+			channel.Members = membersToKeep // keep only members that can post
```

Comment:
Does the status-desktop client need to adapt to this concept, i.e. only posters on announcement channels? If so, this would be a breaking protocol change, as it goes over the wire and will eventually reach older clients.

Reply:
seems so

https://github.com/status-im/status-go/blob/develop/protocol/communities/community.go#L228

Reply:
> seems so
> 
> https://github.com/status-im/status-go/blob/develop/protocol/communities/community.go#L228

Confusing naming :sweat_smile: 

Please make sure announcement channels work, also for custom members (not only privileged ones) being posters :pray: